client/state/jetpack/credentials/reducer.js

```diff
@@ -63,6 +66,21 @@ export const updateRequestStatus = keyedReducer( 'siteId', ( state, { type } ) =
 	return state;
 } );
 
+export const testRequestStatus = keyedReducer( 'siteId', ( state, { type, role, testResult } ) => {
+	switch ( type ) {
+		case JETPACK_CREDENTIALS_TEST:
+			return { [ role ]: 'pending' };
+
+		case JETPACK_CREDENTIALS_TEST_SUCCESS:
+			return { [ role ]: testResult ? 'valid' : 'invalid' };
+
+		case JETPACK_CREDENTIALS_TEST_FAILURE:
+			return { [ role ]: 'failed' };
```

Comment:
What's the difference between failed and invalid, and what difference do they make when it comes to implementation? Can we ideally hold a single failure state?

Reply:
Well, technically the failure state is when we have an internal issue connecting to VP/Transport Server. For this implementation, I think we can consider failure as invalid because we will ask again for credentials.

Reply:
I'll make the change and let you know.